src/Types/Filtering.hs

```diff
@@ -44,5 +48,8 @@ instance Show FunctionSignature where
         constraintsExpr = (intercalate ", " . map show) constraints
         argsExpr = (intercalate " -> " . map show) (argsType ++ [returnType])
 
-defaultTimeout = 3000000 :: Int
-defaultNumChecks = 10 :: Int
\ No newline at end of file
+data SampleResult =
```

Comment:
This looks like it's encoding the same thing as a `Maybe Sample`, I think that would be easier to work with--you'll get some functions for free.